hydra_pspec/pspec.py

```diff
@@ -1,12 +1,57 @@
 import numpy as np
 import scipy as sp
+from scipy.interpolate import interp1d
 from scipy.stats import invgamma
 
 from multiprocess import Pool, current_process
 from . import utils
 import time
 
 
+def inversion_sample_invgamma(alpha, beta, prior_min, prior_max, ngrid=1000):
+    """
+    Draw a sample from an inverse gamma distribution between prior bounds using
+    inversion sampling.
+    
+    This works by sampling the cdf of the inverse gamma distribution 
+    on a (logarithmic) grid and then interpolating to convert a uniform 
+    random draw into a random draw with the correct PDF.
+
+    Parameters:
+        alpha (float):
+            Inverse gamma alpha parameter.
+        beta (float):
+            Inverse gamma beta (scale) parameter.
+        prior_min (float):
+            Minimum of the prior range.
+        prior_max (float):
+            Maximum of the prior range.
+        ngrid (int):
+            Number of sample points to use for interpolator.  Defaults to 1000.
+
+    Returns:
+        sample (float):
+            Sample drawn from the inverse gamma distribution between the 
+            specified prior bounds.
+    """
+    # Sample cdf logarithmically between provided prior bounds
```

Comment:
Perhaps add an assert to make sure log10(prior_min) and prior_max are valid

Reply:
By rights, we should also check that prior_min < prior_max...

Reply:
Added (I think) all the necessary checks.  Let me know if I missed anything.